api/base/utils.py

```diff
@@ -149,7 +149,7 @@ def default_node_list_queryset(model_cls):
 
 def default_node_permission_queryset(user, model_cls):
     assert model_cls in {Node, Registration}
-    if user.is_anonymous:
+    if user is None or user.is_anonymous:
```

Comment:
user is passed as None here: https://github.com/CenterForOpenScience/osf.io/pull/8437/files#diff-fda6db2343d9c13b03db5dd94083d909R153, so it is because of tests, but I think this is actually the correct behavior,